File: ios/RTLViewManager.swift
import React
import RTLSdk
import UIKit

@objc(RTLViewManager)
final class RTLViewManager: RCTViewManager {
    override static func requiresMainQueueSetup() -> Bool {
        true
    }

    override func view() -> UIView! {
        let view = RTLSdk.shared.createWebView()
        view.isUserInteractionEnabled = true
        return view
    }
}

File: android/src/main/java/com/affinaloyalty/rtlsdk/reactnative/RTLViewManager.kt
package com.affinaloyalty.rtlsdk.reactnative

import android.view.ViewGroup
import android.widget.FrameLayout
import com.affinaloyalty.rtlsdk.RTLSdk
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext

class RTLViewManager : SimpleViewManager<FrameLayout>() {
    override fun getName(): String = NAME

    override fun createViewInstance(reactContext: ThemedReactContext): FrameLayout {
        return FrameLayout(reactContext).apply {
            layoutParams = ViewGroup.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )

            val webViewContext = reactContext.currentActivity ?: reactContext
            val webView = RTLSdk.getInstance().createWebView(webViewContext)
            addView(
                webView,
                FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.MATCH_PARENT,
                    FrameLayout.LayoutParams.MATCH_PARENT
                )
            )
        }
    }

    companion object {
        const val NAME = "RTLView"
    }
}
